code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -68,6 +68,12 @@
 	. = ..()
 	if(flag)
 		return // too close
+	if(user?.mind?.martial_art?.no_guns)
```

Comment:
I don't quite see a case where `user` would be null in all honesty. I know they guarded it below as well but now it will cause an exception in `HAS_TRAIT` if it's null.
You could rewrite this proc so it checks the user beforehand. Or just leave the check out

Reply:
done